### DISCUSSION TITLE
Migrating Development Efforts to TanStack Form

### DISCUSSION THREAD

**Discussion by crutchcorn** [houseform/houseform#121] [Announcements]
Welp - guess the news is out!

I'm working with Tanner Linsley (of React Query fame) on TanStack Form

<img width="451" alt="" src="https://github.com/houseform/houseform/assets/9100169/e8f6576f-8b04-4fb7-9783-cf15641d7e52">
<br/><br/><br/>
<a href="https://twitter.com/tannerlinsley/status/1696619675990540316"><img width="451" alt="" src="https://github.com/houseform/houseform/assets/9100169/d48a22fd-4a63-4667-8654-27879e69f59f"></a>

This will replace HouseForm as my primary effort of fixing forms in the React (and Vue) ecosystems.

# What does this mean for HouseForm?

Large-scale development on HouseForm will be paused. While we'll still accept pull requests, I will not be putting my efforts into anything substantial on HouseForm except bug fixes and PR reviews.

That said, my company is still actively using and building on top of HouseForm and will continue to do so until TanStack Form sees a v1 release. If you're using HouseForm in production today, rest assured that the project is not abandoned in terms of bugfixes and minor improvements at this time.

When TanStack v1 releases, I will have a step-by-step migration guide, most features from HouseForm ported over to TanStack, and maybe even a codemod.

Once v1 of TanStack Form launches, HouseForm will:

- Write a blog post formally introducing TanStack Form, [just like we did for HouseForm](https://dev.to/crutchcorn/formik-works-great-heres-why-i-wrote-my-own-591m)
- Write a migration guide from HouseForm to TanStack Form
- Shut down our Discord server
- Archive the GitHub repository

# Why TanStack Form?

While I still strongly believe in HouseForm's Field-first methodology (which TanStack Form shares), TanStack Form improves on HouseForm by:

- Being framework agnostic
  - I'll be launching a React and Vue version for TanStack Form v1
- Being validation adapter agnostic
  - Day 1 support for:
    - Zod
    - Yup
    - [Valibot](https://github.com/fabian-hiller/valibot)
- Introducing a `useForm` hook that fixes TypeScript issues with `<Form/>` and `<Field/>` linking

We're still in early days, so docs needs some TLC, but let me know what you think of the syntax:

[TanStack Form Homepage](https://tanstack.com/form/latest)

# What does this mean for TanStack Form?

We've done this before! Not only have I consumed two other major form libraries in projects (Formik and React Hook Form), but I've written HouseForm to be in a stable place. I think I have a good idea of what features I want to see in TanStack Form from HouseForm.

Namely, I want to make mirror work on the following features:

- [houseform#45](https://github.com/houseform/houseform/issues/45) -> [tanstack/form#417](https://github.com/TanStack/form/issues/417)
- [houseform#60](https://github.com/houseform/houseform/issues/60) -> [tanstack/form#418](https://github.com/TanStack/form/issues/418)
- [houseform#86](https://github.com/houseform/houseform/issues/86) -> [tanstack/form#419](https://github.com/TanStack/form/issues/419)
- [houseform#105](https://github.com/houseform/houseform/issues/105) (already done in TanStack Form!)

This means that you can expect to see these features in TanStack Form at some point in the future 😊

# How can I help?

Want to help us with TanStack Form? You're amazing! 😄

We need help on everything from:

- Testing & bug reports
- Feature requests
- Docs
- Issue triage/creation
- Integration Tests
- TypeScript Type Tests
- Vue support
- Benchmarking

If any of this seems interesting, [join our Discord](https://tlinz.com/discord) and ping me in the `#form` channel - we'll figure out how to divvy up the work and assign it out! 😄 

Thank you for taking an interest in our work and for being a supportive member of the community. Until next time ✌️

**Comment by babycourageous** [comment]
This is so very exciting! A match made in heaven!

**Comment by gmwill934** [comment]
oh yeah!

**Comment by AFusco** [comment]
Great work! Is there an ETA for tanstack form v1?
